app/routes/create-agreement/soil-assessment.js

```diff
@@ -7,6 +7,9 @@ module.exports = [{
   path: '/create-agreement/soil-assessment',
   options: {
     handler: async (request, h) => {
+      await cache.update('agreement', request.yar.id, {
+        agreementLength: 'rolling'
+      })
```

Comment:
Maybe this should move elsewhere as feels a bit hidden in the journey here.  Maybe when the agreement is submitted when we add the id?  also add the length there rather than tie it to soil assessment?